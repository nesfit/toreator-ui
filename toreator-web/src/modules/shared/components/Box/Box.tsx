import styled from "styled-components";
import {
  alignSelf,
  color,
  display,
  flex,
  fontSize,
  fontWeight,
  justifyContent,
  maxWidth,
  minWidth,
  space,
  textAlign,
  width,
} from "styled-system";

const Box: any = styled.div`
  ${space}
  ${width}
  ${fontSize}
  ${fontWeight}
  ${color}
  ${flex}
  ${alignSelf}
  ${maxWidth}
  ${minWidth}
  box-sizing: border-box;
  ${display}  
  ${justifyContent}  
  ${textAlign}  
`;

export default Box;
