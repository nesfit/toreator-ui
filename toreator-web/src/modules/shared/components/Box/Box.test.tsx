import {shallow} from "enzyme";
import * as React from "react";
import Box from "./Box";

describe("<Box/>", () => {
  test("default", () => {
    const wrapper = shallow(<Box />);
    expect(wrapper).toMatchSnapshot();
  });
});
