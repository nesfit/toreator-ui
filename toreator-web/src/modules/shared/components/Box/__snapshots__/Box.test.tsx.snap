// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box/> default 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  box-sizing: border-box;
  ",
          [Function],
          "  
  ",
          [Function],
          "  
  ",
          [Function],
          "  
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
